bondjimbond/islandora_westvault
# Preserve child objects based on parent settings

It works!

Latest changes:

- Preservation method set in PRESERVATION datastream
- Default value for Collection preservation method set according to existing PRESERVATION datastream
- New and existing Child objects are preserved depending on the preservation method, as recorded in the PRESERVATION datastream
- If a collection is already preserved and the preservation method is changed, the existing PRESERVATION datastream is purged and replaced by a new one with the desired PRESERVATION method.
- Collection objects with PRESERVATION datastreams are ignored - partly because bag-it isn't working with them and that causes problems; also because we are preserving individual objects and not entire collections here.

To test:
- Create a fresh collection with some fresh content (islandora_scg is good for this)
- Check the PRESERVATION box for one or two items; leave the rest alone
- Preserve the collection; set to "new children". Check teh PRESERVATION datastream to make sure that it contains `<preservationMethod>newchildren</preservationMethod>`
- Make sure no existing objects have been changed
- Preserve a new object; it should have a PRESERVATION datastream
- Manage the collection again, change the method to "current children only"; check the collection's PRESERVATION datastream to confirm the change.
- All children should now have a PRESERVATION datastream
- Ingest a new object; should not get a datastream
- Change collection's preservation method again to "All children"
- The object you just ingested should now have a PRESERVATION datastream
- Ingest another object; it should get a PRESERVATION datastream
